web/pages/leaderboards.tsx

```diff
@@ -30,31 +40,68 @@ export default function Leaderboards(props: {
     topTraders: [],
     topCreators: [],
   }
-  const { topTraders, topCreators } = props
+
+  const [topTradersState, setTopTraders] = useState(props.topTraders)
+  const [topCreatorsState, setTopCreators] = useState(props.topCreators)
+  const [isLoading, setLoading] = useState(false)
+  const [period, setPeriod] = useState<LeaderboardPeriod>('weekly')
+
+  useEffect(() => {
+    setLoading(true)
+    queryTopCreators(period).then((res) => {
+      setTopTraders(res.props.topTraders as User[])
+      setTopCreators(res.props.topCreators as User[])
+      setLoading(false)
+    })
+  }, [period])
 
   return (
     <Page margin>
+      <Title text={'Leaderboards'} className={'hidden md:block'} />
+      <Col className="gap-4 text-center">
+        <select
+          onChange={async (e) => {
+            setPeriod(e.target.value as LeaderboardPeriod)
+          }}
+          value={period}
+          className="!select !select-bordered max-w-xs"
+        >
+          <option value="allTime">All Time</option>
+          <option value="monthly">Monthly</option>
+          <option value="weekly">Weekly</option>
```

Comment:
Great. Reminder to remove these options before merging. (Maybe add a TODO to add them back in at the relevant time?)

Reply:
Definitely!